app/src/androidTest/java/fr/free/nrw/commons/WelcomeActivityTest.kt

```diff
@@ -34,4 +35,43 @@ class WelcomeActivityTest {
                     .check(matches(not(isDisplayed())))
         }
     }
+
+    @Test
+    fun ifSkipButtonClicked() {
+        if (!ConfigUtils.isBetaFlavour()) {
+            onView(withId(R.id.finishTutorialButton))
+                    .perform(ViewActions.click())
+            assert(activityRule.activity.isDestroyed)
+        }
+    }
+
+    @Test
+    fun ifSwiped() {
+        if (!ConfigUtils.isBetaFlavour()) {
+            onView(withId(R.id.welcomePager))
+                    .perform(ViewActions.swipeLeft())
+            assert(true)
+            onView(withId(R.id.welcomePager))
+                    .perform(ViewActions.swipeRight())
+            assert(true)
+        }
+    }
+
+    @Test
+    fun ifEndedTutorial() {
```

Comment:
Maybe rename this to something that better describes it - e.g. `testSwipingWholeTutorial`. Reason the methods already there we prefixed with 'if' was just that they are special cases (i.e. if beta, then it should show the skip button).